hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -61,19 +66,31 @@ public class TpcServerBootstrap {
     private final boolean enabled;
     private final Map<Eventloop, Supplier<? extends ReadHandler>> readHandlerSuppliers = new HashMap<>();
     private final List<AsyncServerSocket> serverSockets = new ArrayList<>();
+    private final Config config;
     private volatile List<Integer> clientPorts;
 
     public TpcServerBootstrap(NodeEngineImpl nodeEngine) {
         this.nodeEngine = nodeEngine;
         this.logger = nodeEngine.getLogger(TpcServerBootstrap.class);
         this.ss = (InternalSerializationService) nodeEngine.getSerializationService();
-        this.enabled = Boolean.parseBoolean(getProperty("hazelcast.tpc.enabled", "false"))
-                || nodeEngine.getConfig().getAltoConfig().isEnabled();
-        logger.info("TPC: " + (enabled ? "enabled" : "disabled"));
+        this.config = nodeEngine.getConfig();
+        this.enabled = calculateEnabled();
         this.thisAddress = nodeEngine.getThisAddress();
         this.tpcEngine = newTpcEngine();
     }
 
+    private boolean calculateEnabled() {
```

Comment:
Don't call this 'calculate'. It isn't mathematics.

Reply:
This defaulting logic should be part of ClusterProperty definition

```
 public static final HazelcastProperty NETWORK_STATS_REFRESH_INTERVAL_SECONDS
            = new HazelcastProperty("hazelcast.network.stats.refresh.interval.seconds", 3, SECONDS);
```

So when you define the property, it defaults to 3 second in this case. ANd then the whole 'calculateEnabled' probably completely disappears since it can all be done by the HazelcastProperty.

Reply:
Well than you duplicate default value. It will exist in `AltoConfig` and `ClusterProperty`.

Reply:
I think this is where the complexity comes from. Normally we either have a setting following the path of ClusterProperty, or explicit Config property. But in this case we have both.

Reply:
I will export the default value into public static final field and use that in ClusterProperty. Does it sound good enough?

Reply:
It is fine for now.